rotkehlchen/rotkehlchen.py

```diff
@@ -1085,16 +1085,17 @@ def setup_exchange(
             )
         return is_success, msg
 
-    def query_periodic_data(self) -> dict[str, Union[bool, list[str], Timestamp]]:
+    def query_periodic_data(self) -> dict[str, Union[bool, dict[str, list[str]], Timestamp]]:
         """Query for frequently changing data"""
-        result: dict[str, Union[bool, list[str], Timestamp]] = {}
+        result: dict[str, Union[bool, dict[str, list[str]], Timestamp]] = {}
 
         if self.user_is_logged_in:
             with self.data.db.conn.read_ctx() as cursor:
                 result['last_balance_save'] = self.data.db.get_last_balance_save_time(cursor)
-                result['connected_eth_nodes'] = [node.name for node in self.chains_aggregator.ethereum.node_inquirer.get_connected_nodes()]  # noqa: E501
-                result['connected_optimism_nodes'] = [node.name for node in self.chains_aggregator.optimism.node_inquirer.get_connected_nodes()]  # noqa: E501
-                result['connected_polygon_pos_nodes'] = [node.name for node in self.chains_aggregator.polygon_pos.node_inquirer.get_connected_nodes()]  # noqa: E501
+                connected_nodes = {}
+                for supported_chain in self.chains_aggregator.iterate_supported_chains():
+                    connected_nodes[supported_chain.to_name()] = [node.name for node in self.chains_aggregator.get_evm_manager(supported_chain).node_inquirer.get_connected_nodes()]  # noqa: E501
```

Comment:
if you do what I told you, the loop variable would be the evm manager, and from that you can get the chain name by doing `evm_manager.node_inquirer.chain_name`